src/GeniusYield/Transaction.hs

```diff
@@ -488,4 +492,46 @@ makeTransactionBodyAutoBalanceWrapper collaterals ss eh pp ps utxos body changeA
         {- Technically, this doesn't compare with the _final_ tx size, because of signers that will be
         added later. But signing witnesses are only a few bytes, so it's unlikely to be an issue -}
         Left (BuildTxSizeTooBig maxTxSize txSize)
-    first BuildTxCborSimplificationError $ simplifyGYTxBodyCbor $ txBodyFromApi txBody
+
+    collapsedBody <- first BuildTxCollapseExtraOutError $ collapseExtraOut extraOut txBodyContent outsOld
+
+    first BuildTxCborSimplificationError $ simplifyGYTxBodyCbor $ txBodyFromApi collapsedBody
+
+
+{- | Collapses the extra out generated in the last step of tx building into
+    another change output (If one exists)
+-}
+collapseExtraOut
+  :: Api.TxOut Api.S.CtxTx Api.S.BabbageEra
+  -- ^ The extra output generated by makeTransactionBodyAutoBalance
+  -> Api.TxBodyContent Api.S.BuildTx Api.S.BabbageEra
+  -- ^ The body content generted by makeTransactionBodyAutoBalance
+  -> [GYTxOut v]
+  -- ^ The outputs that we don't want to modify
+  -> Either Api.S.TxBodyError (Api.S.TxBody Api.S.BabbageEra)
+  -- ^ The updated body with the collapsed outputs
+collapseExtraOut apiOut@(Api.TxOut outAddr outVal _ _) bodyContent@Api.TxBodyContent {txOuts} outsOld
+ | Api.txOutValueToLovelace outVal == 0 = Api.S.createAndValidateTransactionBody bodyContent
+ | otherwise =
+    case delete apiOut cOuts of
+        [] -> Api.S.createAndValidateTransactionBody bodyContent
+        -- sOut == selected Out == A cOut != eOut that will merge with eOut
+        (sOut@(Api.TxOut sOutAddr sOutVal sOutDat sOutRefScript):_) ->
+            let
+
+            nOutVal = case sOutVal of
+                Api.TxOutAdaOnly oasie l -> Api.TxOutAdaOnly oasie $ Api.txOutValueToLovelace outVal <> l
+                Api.TxOutValue masie val -> Api.TxOutValue masie $ Api.txOutValueToValue outVal <> val
+
+            -- nOut == new Out == The merging of both eOut and sOut
+            nOut = Api.TxOut sOutAddr nOutVal sOutDat sOutRefScript
+            -- nOuts == new Outs == The new list of all old outs mins eOut and sOut plus the nOut
+            nOuts = (txOuts \\ [apiOut, sOut]) ++ [nOut]
```

Comment:
This can be problematic, what if `apiOut` or `sOut` turn out to be same as one of user's mentioned skeleton output.

Reply:
`apiOut` is the one added by the Cardano Api. And `sOut` is the head of `cOuts` which is the change outputs generated by the Coin selection. 
The complete flow is as follows:

* We get all outs from the txBodyContent (`txOuts`)
* We delete the outputs mentioned in the skeleton (`outsOld`) and are left with `outsNew`
* We filter the outputs with the same address than the output generated by the cardanoApi and are left with `cOuts`
* We take out the output generated by the cardanoApi (`apiOut`) from `cOuts`. 
    * If that list is empty, we can't collapse the extra output
    * If the list is not empty, we chose the first element to collapse into (`sOut`)
        * We get the new value by adding the value of sOut and apiOut
        * We generate a new output (nOut) by copying all info from sOut but using the new, updated value
        * We delete the outputs that were collapsed from the list of outputs and add the new, collapsed output to the list.
        * We return the body with the updated list of outputs.

Reply:
Now that I wrote the whole flow I realized that the oldOuts could be adjusted by the coin selection step, so the solution proposed in the other comment is looking better and better.

Reply:
Maybe you misunderstood me or I am misunderstanding you. I just meant, when you do `nOuts = (txOuts \\ [apiOut, sOut])`, we are suppose to remove actual `apiOut` & `sOut` and not any output which may happen to match with them. Suppose `txOuts` was `[o1, o2, o3, c1 = o1, c2, c3]` where $o_i$ are user specified outputs and $c_j$ are change outputs. Then you may remove first $o_1$ instead of later desired $c_1$. Here first change output `sOut` turned out to be same as one of user specified skeleton output.

Reply:
ah, ok. I understand the problem now. I though about that but I initially though that if o1 and  c1 are equal, they have the same address, value, datum and ref script it doesn't matter which one is deleted. But In that case the index for each output will be different, so it's not acceptable.

Please check the new implementation which should address both concerns.